accounts/authentication.py

```diff
@@ -40,8 +41,6 @@ def authenticate(self, request):
                 headers=headers,
                 data=body,
             )
-            if platform_request.status_code != 200:  # Access token is invalid
```

Comment:
@pawalt Theoretically, `PlatformAuthentication` should not have been changed in this PR because `PlatformAuthentication` is only used when we are talking to a product through a user.

Why would we need to delete this part? Wouldn't it just fail later on if the status code is 200?